applications/server/models/tables.py

```diff
@@ -75,6 +59,25 @@
                 # See above.
                 )
 
+#########################################################################
+
+# Procedure Harness Tables
+
+# with this new split table definition it makes sense to just use the automatic id in this table as the procedure_id
+db.define_table('procedures',
+                #Field('procedure_id', 'bigint', required=True),  # key
+                Field('user_email', 'string', required=True),
+                Field('device_id', 'string', required=True),
+                Field('name', 'string')  # Name of procedure
+                )
+
+db.define_table('procedure_revisions',
+                Field('procedure_id', 'bigint', required=True),  # key
+                Field('procedure_data', 'text', required=True),  # Actual code for procedure - is check IS_LENGTH(65536) ok?
+                # Otherwise use string and specifiy larger length
+                Field('last_update', 'datetime', default=datetime.utcnow(), required=True),
+                Field('stable_version', 'boolean', required=True) # True for stable False for not stable
```

Comment:
I like to call boolean fields like this with names like is_stable, so I know that they are boolean.  stable_version gives me the idea that it contains a revision id.